<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ur3_robotiq85_gripper.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur3" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- measured from model -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/base.dae"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0030531654454" ixy="0.0" ixz="0.0" iyy="0.0030531654454" iyz="0.0" izz="0.005625"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1519"/>
    <axis xyz="0 0 1"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/shoulder.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0080931634294" ixy="0.0" ixz="0.0" iyy="0.0080931634294" iyz="0.0" izz="0.005625"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.1198 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/upperarm.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.42"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.121825"/>
      <inertia ixx="0.0217284832211" ixy="0.0" ixz="0.0" iyy="0.0217284832211" iyz="0.0" izz="0.00961875"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.0925 0.24365"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/forearm.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.26"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.106625"/>
      <inertia ixx="0.00654680644378" ixy="0.0" ixz="0.0" iyy="0.00654680644378" iyz="0.0" izz="0.00354375"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.21325"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/wrist1.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.8"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.002084999166" ixy="0.0" ixz="0.0" iyy="0.002084999166" iyz="0.0" izz="0.00225"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.08505 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/wrist2.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.8"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.002084999166" ixy="0.0" ixz="0.0" iyy="0.002084999166" iyz="0.0" izz="0.00225"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.08535"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.5" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/visual/wrist3.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur3/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.35"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000912187135125" ixy="0.0" ixz="0.0" iyy="0.000912187135125" iyz="0.0" izz="0.000984375"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0819 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0819 0"/>
    <parent link="wrist_3_link"/>
    <child link="tool0"/>
  </joint>
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
  <!-- Robotiq Coupler -->
  <!--  + Height added by the coupler: 8mm -->
  <!--  + Reference frame: at the middle (4mm) -->
  <link name="robotiq_coupler">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/robotiq_85_coupler.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/robotiq_85_coupler.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="2.073e-05   1.45286e-03  -1.1049e-03"/>
      <mass value="0.168"/>
      <inertia ixx="6.69695624e-05" ixy="5.21511788e-09" ixz="-2.74383009e-08" iyy="7.85088161e-05" iyz="5.41105193e-07" izz="1.41819717e-04"/>
    </inertial>
  </link>
  <joint name="robotiq_coupler_joint" type="fixed">
    <origin rpy="0 0 -1.57079632679" xyz="0 0 0.004"/>
    <parent link="tool0"/>
    <child link="robotiq_coupler"/>
  </joint>
  <!-- Equivalent to the OpenRAVE manipulator denso_robotiq_85_gripper -->
  <link name="ur3_robotiq_85_gripper"/>
  <joint name="manipulator_dummy_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.1441"/>
    <parent link="robotiq_coupler"/>
    <child link="ur3_robotiq_85_gripper"/>
  </joint>
  <joint name="robotiq_85_base_joint" type="fixed">
    <parent link="robotiq_coupler"/>
    <child link="robotiq_85_base_link"/>
    <origin rpy="0 -1.57079632679 3.14159265359" xyz="0 0 0.004"/>
  </joint>
  <link name="robotiq_85_base_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_base_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_base_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.636951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000380" ixy="0.000000" ixz="0.000000" iyy="0.001110" iyz="0.000000" izz="0.001171"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_knuckle_joint" type="revolute">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0.0 0.0" xyz="0.05490451627 0.03060114443 0.0"/>
    <limit effort="1000" lower="0.0" upper="0.80285" velocity="0.5"/>
  </joint>
  <joint name="robotiq_85_right_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.05490451627 -0.03060114443 0.0"/>
    <limit effort="1000" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint"/>
  </joint>
  <link name="robotiq_85_left_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyy="0.000001" iyz="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyy="0.000001" iyz="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_finger_joint" type="fixed">
    <parent link="robotiq_85_left_knuckle_link"/>
    <child link="robotiq_85_left_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <joint name="robotiq_85_right_finger_joint" type="fixed">
    <parent link="robotiq_85_right_knuckle_link"/>
    <child link="robotiq_85_right_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <link name="robotiq_85_left_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_finger_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyy="0.000021" iyz="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_finger_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyy="0.000021" iyz="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0.0 0.0" xyz="0.06142 0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" offset="0"/>
  </joint>
  <joint name="robotiq_85_right_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.06142 -0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" offset="0"/>
  </joint>
  <link name="robotiq_85_left_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyy="0.000005" iyz="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyy="0.000005" iyz="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_left_inner_knuckle_link"/>
    <child link="robotiq_85_left_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821 0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" multiplier="-1"/>
  </joint>
  <joint name="robotiq_85_right_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_right_inner_knuckle_link"/>
    <child link="robotiq_85_right_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821  0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" multiplier="-1"/>
  </joint>
  <link name="robotiq_85_left_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyy="0.000005" iyz="0.000000" izz="0.000006"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyy="0.000005" iyz="0.000000" izz="0.000006"/>
    </inertial>
  </link>
</robot>
